### ISSUE TITLE
`placeholder "out"` does not work with `passAsFile`

### ISSUE THREAD

**Issue by adelbertc** [NixOS/nix#2199]
`default.nix`:

```
let
  pkgs = import <nixpkgs> { };
in
  pkgs.writeShellScriptBin "hello" ''
    echo ${placeholder "out"}
  ''
```

Test:
```
~/dev/nix-experiments/placeholder-out ⊢ nix-build
/nix/store/0aamnh6hdr9qh34nzcl5r6anaj1zb5f8-hello
~/dev/nix-experiments/placeholder-out ⊢ ./result/bin/hello 
/1rz4g4znpzjwh1xymhjpm42vipw92pr73vdgl6xs1hycac8kf2n9
```

The expectation is `${placeholder "out"}` gives the Nix store path of the script, but:

1. Only what appears to be a suffix is given
2. Even if you prefix with `/nix/store` the path is bogus

Initial guess is this is due to the use of `passAsFile` by `writeShellScriptBin`

**Comment by shlevy** [comment]
Since placeholder is Nix 2.0 only, I wonder if we can just use `__structuredAttrs` instead... Does `placeholder` work there?

**Comment by adelbertc** [comment]
@shlevy Was that directed at me? Reading up on `__structuredAttrs` not sure what you mean

**Comment by jtojnar** [comment]
Duplicate of https://github.com/NixOS/nix/issues/1846

**Comment by FlorianFranzen** [comment]
Should have been fixed by #3463, so probably can be closed.
